routing/router.go

```diff
@@ -2699,11 +2698,23 @@ func (r *ChannelRouter) sendToRoute(htlcHash lntypes.Hash, rt *route.Route,
 // carry out its execution. After restarts, it is safe, and assumed, that the
 // router will call this method for every payment still in-flight according to
 // the ControlTower.
-func (r *ChannelRouter) sendPayment(feeLimit lnwire.MilliSatoshi,
-	identifier lntypes.Hash, timeout time.Duration,
-	paySession PaymentSession,
+func (r *ChannelRouter) sendPayment(ctx context.Context,
+	feeLimit lnwire.MilliSatoshi, identifier lntypes.Hash,
+	paymentAttemptTimeout time.Duration, paySession PaymentSession,
 	shardTracker shards.ShardTracker) ([32]byte, *route.Route, error) {
 
+	// If the user provides a timeout, we will additionally wrap the context
+	// in a deadline.
+	cancel := func() {}
+	if paymentAttemptTimeout > 0 {
+		timeout := time.Now().Add(paymentAttemptTimeout)
+		ctx, cancel = context.WithDeadline(ctx, timeout)
```

Comment:
we can call `defer cancel()` here to avoid declaring it?

Reply:
I replaced the `WithDeadline` with `WithTimeout`, but I still have to declare `cancel` since I can't re-declare `ctx` here.

Reply:
oh that's right, yea in this case it's fine as-is.